framework/rendering/render_frame.cpp

```diff
@@ -180,10 +185,17 @@ DescriptorSet &RenderFrame::request_descriptor_set(DescriptorSetLayout &descript
 {
 	assert(thread_index < thread_count && "Thread index is out of bounds");
 
-	auto &descriptor_pool = request_resource(device, nullptr, *descriptor_pools.at(thread_index), descriptor_set_layout);
+	auto &descriptor_pool = request_descriptor_pool(descriptor_set_layout, thread_index);
 	return request_resource(device, nullptr, *descriptor_sets.at(thread_index), descriptor_set_layout, descriptor_pool, buffer_infos, image_infos);
 }
 
+DescriptorPool &RenderFrame::request_descriptor_pool(DescriptorSetLayout &descriptor_set_layout, size_t thread_index)
```

Comment:
`descriptor_set_layout` isn't modified here - should be a `const &`

Reply:
Ended up deleting the method due to another comment, but noticed that `request_descriptor_set` and `descriptor_set` itself have the same issue, so added `const` there as well.